klient/k8s/res/resource.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package res
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	cr "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/e2e-framework/klient/k8s"
+)
+
+type Resources struct {
+	// config is the rest.Config to talk to an apiserver
+	config *rest.Config
+
+	// scheme will be used to map go structs to GroupVersionKinds
+	scheme *runtime.Scheme
+}
+
+func Res(cfg *rest.Config) *Resources {
+	if cfg == nil {
+		fmt.Println("must provide rest.Config")
+		return nil
+	}
+
+	res := &Resources{
+		config: cfg,
+		scheme: scheme.Scheme,
+	}
+
+	return res
+}
+
+// New wrapper fumction for controller runtimr client New()
+func (r *Resources) New() cr.Client {
+	client, err := cr.New(r.config, cr.Options{Scheme: r.scheme})
+	if err != nil {
+		// TODO: log error and return
+		return nil
+	}
+
+	return client
+}
+
+func (r *Resources) Get(ctx context.Context, name, namespace string, obj k8s.Object) error {
+	return r.New().Get(ctx, cr.ObjectKey{Namespace: namespace, Name: name}, obj)
```

Comment:
When the `r` (Resources) has the instance of the cr.Client, this should be changed to just used the stored client
```go
r.client.Get(ctx, cr.ObjectKey{Namespace: namespace, Name: name}, obj)
```